netty/src/test/java/io/grpc/netty/NettyServerTest.java

```diff
@@ -91,4 +102,83 @@ public void getPort_notStarted() throws Exception {
 
     assertThat(ns.getPort()).isEqualTo(-1);
   }
+
+  @Test(timeout = 60000)
+  public void childChannelOptions() throws Exception {
+    final int receiveBufferSize = 2097169;
+    final int sendBufferSize = 2097211;
+
+    Map<ChannelOption<?>, Object> channelOptions = new HashMap<ChannelOption<?>, Object>();
+
+    channelOptions.put(ChannelOption.SO_RCVBUF, receiveBufferSize);
+    channelOptions.put(ChannelOption.SO_SNDBUF, sendBufferSize);
+
+    final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+    NettyServer ns = new NettyServer(
+        new InetSocketAddress(9999),
+        NioServerSocketChannel.class,
+        channelOptions,
+        null, // no boss group
+        null, // no event group
+        new ProtocolNegotiators.PlaintextNegotiator(),
+        Collections.<ServerStreamTracer.Factory>emptyList(),
+        TransportTracer.getDefaultFactory(),
+        1, // ignore
+        1, // ignore
+        1, // ignore
+        1, // ignore
+        1, // ignore
+        1, 1, // ignore
+        1, 1, // ignore
+        true, 0); // ignore
+    ns.start(new ServerListener() {
+      @Override
+      public ServerTransportListener transportCreated(ServerTransport transport) {
+        Channel channel = ((NettyServerTransport)transport).channel();
+        assertThat(channel.config().getOption(ChannelOption.SO_RCVBUF))
+            .isEqualTo(receiveBufferSize);
+        assertThat(channel.config().getOption(ChannelOption.SO_SNDBUF))
+            .isEqualTo(sendBufferSize);
+
+        countDownLatch.countDown();
+
```

Comment:
nit, if an error happens here (like an assertion fails), the test thread wont find out about it.  On receiving these values you should set them in an atomic integer or something, and then check for the test thread.

Reply:
if an error happens here, assertion will fail. And easy It's for developers for figure out what happened. :-)

Reply:
travis-ci node will overwrite receiveBufferSize and sendBufferSize?